src/shared/components/TextArea/TextArea.tsx

```diff
@@ -1,84 +1,42 @@
 import React, { forwardRef, useState } from 'react'
-import { getVariant } from '../InputBase'
-import { HelperText, HelperTextCount, HelperTextsWrapper, StyledTextArea, TextAreaWrapper } from './TextArea.style'
+import InputBase, { InputBaseProps } from '../InputBase'
+import { StyledTextArea } from './TextArea.style'
 
 export type TextAreaProps = {
   name?: string
   placeholder?: string
   onChange?: (event: React.ChangeEvent<HTMLTextAreaElement>) => void
   value?: string
-  maxLength?: number
   className?: string
-  helperText?: string
-  warning?: boolean
-  error?: boolean
   rows?: number
   spellcheck?: boolean
-}
+} & InputBaseProps
+
 const TextAreaComponent: React.ForwardRefRenderFunction<HTMLTextAreaElement, TextAreaProps> = (
-  {
-    onChange,
-    name,
-    placeholder = 'Something should be here',
-    value,
-    maxLength,
-    className,
-    rows = 10,
-    helperText = '\u00A0',
-    warning,
-    error,
-    spellcheck = true,
-  },
+  { onChange, name, placeholder, value, className, rows = 5, disabled, spellcheck = true, ...inputBaseProps },
   ref
 ) => {
-  const [charactersWarning, setCharactersWarning] = useState<'warning' | 'error' | null>(null)
   const [charactersCount, setCharactersCount] = useState(0)
 
   const handleOnChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
-    if (onChange) {
-      onChange(e)
-    }
-
-    if (!maxLength) {
-      return
-    }
-
-    const warningLength = maxLength * 0.8
-    const currentLength = e.target.value.length
-
-    setCharactersCount(currentLength)
-
-    if (currentLength > warningLength) {
-      setCharactersWarning('warning')
-    } else {
-      setCharactersWarning(null)
-    }
-
-    if (currentLength > maxLength) {
-      setCharactersWarning('error')
-    }
+    setCharactersCount(e.target.value.length)
+    onChange?.(e)
   }
 
   return (
-    <TextAreaWrapper className={className}>
+    <InputBase disabled={disabled} charactersCount={charactersCount} {...inputBaseProps}>
       <StyledTextArea
+        className={className}
```

Comment:
Here we pass the `className` to the text area, but in the `TextField`, we pass it to the `InputBase`, we should be consistent with that. I think `InputBase` makes more sense but I'm open to discussion